File: Java/Easy/CompileError/sort-integers-by-the-number-of-1-bits_7.java
class Solution {
    public int[] sortByBits(int[] arr) {
        
        int[] bits = new int[10001];
        for (int i = 0; i < 10001; i++) {
            bits[i] = countBits(i);
        }
        
        Arrays.sort(arr);
        Arrays.sort(arr, (a, b) -> bits[a] - bits[b]);
        return arr;
    }
    
    private int countBits(int n) {
        int count = 0;
        while (n > 0) {
            count += n & 1;
            n >>= 1;
        }
        return count;
    }
}

File: C++/Easy/Accepted/sort-integers-by-the-number-of-1-bits_7.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        
        vector<int> ans;
        sort(arr.begin(), arr.end());
        map<int, vector<int>> m;
        for (auto x : arr) {
            int cnt = 0;
            int y = x;
            while (y) {
                cnt += (y & 1);
                y >>= 1;
            }
            m[cnt].push_back(x);
        }
        for (auto x : m) {
            for (auto y : x.second) {
                ans.push_back(y);
            }
        }
        return ans;
    }
};